perfkitbenchmarker/providers/profitbricks/profitbricks_virtual_machine.py

```diff
@@ -0,0 +1,256 @@
+                'name': self.name,
+                'ram': self.ram,
+                'cores': self.cores,
+                'availabilityZone': self.zone
+            },
+            'entities': {
+                'volumes': {
+                    'items': [
+                        {
+                            'properties': {
+                                'size': self.disk_size,
+                                'name': 'volume1',
+                                'image': self.image,
+                                'bus': 'VIRTIO',
+                                'type': self.profitbricks_disk_type,
+                                'sshKeys': [public_key]
+                            }
+                        }
+                    ]
+                },
+                'nics': {
+                    'items': [
+                        {
+                            'properties': {
+                                'name': 'nic1',
+                                'lan': self.lan_id
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+
+        # Build Server URL
+        url = '%s/datacenters/%s/servers' % (PROFITBRICKS_API, self.dc_id)
+
+        # Provision Server
+        r = util.PerformRequest('post', url, self.header, json=new_server)
+        logging.info('Creating server: %s' % self.name)
+
+        # Parse Required values from response
+        self.server_status = r.headers['Location']
+        response = r.json()
+        self.server_id = response['id']
+
+        # The freshly created server will be in a locked and unusable
+        # state for a while, and it cannot be deleted or modified in
+        # this state. Wait for the action to finish and check the
+        # reported result.
+        if not util.WaitFor(self.server_status, self.header):
+            raise errors.Resource.RetryableCreationError('Server creation '
+                                                         'failed, see log.')
+
+    @vm_util.Retry()
+    def _PostCreate(self):
+        """Get the instance's public IP address."""
+
+        # Poll server to make sure it's been provisioned
+        if util.WaitFor(self.server_status, self.header):
+
+            url = '%s/datacenters/%s/servers/%s?depth=5' % (PROFITBRICKS_API,
+                                                            self.dc_id,
+                                                            self.server_id)
+
+            # Perform Request
+            r = util.PerformRequest('get', url, self.header)
+            response = r.json()
+            nic = response['entities']['nics']['items'][0]
+            self.ip_address = nic['properties']['ips'][0]
+        else:
+            raise errors.Resource.RetryableCreationError('IP parse failed, '
+                                                         'see log.')
+
+    def _Delete(self):
+        """Delete a ProfitBricks Datacenter (Including VM instances)."""
```

Comment:
This deletes the entire data center and everything inside of it (VM, Network Controller, LAN).  If this method only deleted the VM itself, then the user would have to delete the data center/s created by the Perfkit after each iteration which doesn't seem ideal?